### PR TITLE
Bump aws-go-sdk-v2 to v0.30.0

### PR DESCRIPTION
**Pull request by ungureanuvladvictor**
This PR bumps the `aws-go-sdk-v2` dep to v0.30.0.

Between v0.24.0 and v0.25.0 they moved to separate go modules for each service. We use the following packages:
- `github.com/aws/aws-sdk-go-v2/service/ec2`
- `github.com/aws/aws-sdk-go-v2/config` (which was renamed from `external`)
- `github.com/aws/aws-sdk-go-v2/ec2imds` (which was renamed from `ec2metadata`)

The `config` and `ec2imds` modules are versioned separately.

Best way to review is commit by commit. The 1st commit contains the bulk of this given the upgrade was a breaking change (they swapped types and api calls into 2 packages).

While doing this API bump I found a couple of places that we can improve:
- use API [paginators](https://aws.github.io/aws-sdk-go-v2/docs/making-requests/#using-operation-paginators) to make logic simpler for getting collection of objects from the AWS API. For example `describeSubnets` does not paginate so we can expect a customer with a large # of subnets to not be able to use all of them.
- ~bring the code from https://github.com/cilium/cilium/blob/master/pkg/policy/groups/aws/aws.go to use the core logic of the EC2 client that we have in `pkg/aws/ec2`. Currently that codebase constructs its own AWS config + EC2 client with some defaults that might not be correct always.~ -- added in this PR as well.
- integrate the cilium logger into the AWS client via https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/logging/ to get better output if we need to debug internal SDK problems.

### CODE REVIEW

**Comment by ungureanuvladvictor** [comment]
test-me-please

**Comment by tklauser** [comment]
retest-gke

**Comment by tklauser** [comment]
retest-runtime

**Review by tklauser — APPROVED** [review]

**Comment by tklauser** [comment]
retest-runtime

File: pkg/policy/groups/aws/aws.go [hunk 1]
```diff
@@ -96,37 +73,27 @@ func getInstancesIpsFromFilter(ctx context.Context, filter *api.AWSGroup) ([]net
...
-	val := os.Getenv(awsDefaultRegionKey)
-	if val != "" {
-		return val
-	}
-	return awsDefaultRegion
+	return extraxctIPs(result), nil
 }
 
-func awsDumpIpsFromRequest(req *ec2.DescribeInstancesOutput) []net.IP {
+func extraxctIPs(req *ec2.DescribeInstancesOutput) []net.IP {
```
Comment (aanm) [inline pkg/policy/groups/aws/aws.go]: `s/extraxctIPs/extractIPs`

File: pkg/policy/groups/aws/aws.go [hunk 2]
```diff
@@ -96,37 +73,27 @@ func getInstancesIpsFromFilter(ctx context.Context, filter *api.AWSGroup) ([]net
...
-
-// getDefaultRegion returns the given region of the default one.
-// @TODO retrieve the region from aws metadata.
-func getDefaultRegion() string {
-	val := os.Getenv(awsDefaultRegionKey)
-	if val != "" {
-		return val
-	}
-	return awsDefaultRegion
+	return extraxctIPs(result), nil
```
Comment (aanm) [inline pkg/policy/groups/aws/aws.go]: `s/extraxctIPs/extractIPs`

**Review by aanm — CHANGES_REQUESTED** [review]
LGTM, only a minor nit :smile: Won't require a 2nd CI run to fix the typo.

File: pkg/policy/groups/aws/aws.go (hunk 1 above)
Comment (ungureanuvladvictor) [reply pkg/policy/groups/aws/aws.go]: Updated!

File: pkg/policy/groups/aws/aws.go (hunk 2 above)
Comment (ungureanuvladvictor) [reply pkg/policy/groups/aws/aws.go]: Updated!

**Comment by ungureanuvladvictor** [comment]
test-me-please

**Review by christarazi — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
retest-gke

**Comment by tklauser** [comment]
retest-runtime

**Comment by tklauser** [comment]
retest-gke

**Pull request merged**